core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -275,6 +276,29 @@ public boolean isSuggestOnly() {
         return source != null && source.isSuggestOnly();
     }
 
+    /**
+     * Slice this search request into {@code times} separate search requests slicing on {@code field}. Note that the slices are *shallow*
+     * copies of this request so don't change them.
+     */
+    public SearchRequest[] sliceIntoSubRequests(String field, int times) {
+        SearchRequest[] slices = new SearchRequest[times];
+        for (int slice = 0; slice < times; slice++) {
```

Comment:
So, maybe? It is fairly easy to move but moving the *test* is super hard because of `createSearchRequest`. Moving that is fairly 😨 . Would you feel ok with me keeping it here?